helm/templates/promtail/role.yaml

```diff
@@ -0,0 +1,18 @@
+{{- if .Values.rbac.create }}
+apiVersion: rbac.authorization.k8s.io/v1beta1
+kind: Role
+metadata:
+  name: {{ template "promtail.fullname" . }}
+  labels:
+    app: {{ template "promtail.name" . }}
+    chart: {{ .Chart.Name }}-{{ .Chart.Version }}
+    heritage: {{ .Release.Service }}
+    release: {{ .Release.Name }}
+{{- if .Values.rbac.pspEnabled }}
+rules:
+- apiGroups:      ['extensions']
+  resources:      ['podsecuritypolicies']
+  verbs:          ['use']
+  resourceNames:  [{{ template "promtail.fullname" . }}]
```

Comment:
Why is this role needed?  Is this necessary for you to use PSPs?  I'll admit I'm not that familiar with them.

Reply:
At Giant Swarm, we believe the PSP should be enable by default and so it's enable by default on all of our clusters, and also to the customer ones.

However, it can be deactivated via the value param, So I think it's good to keep it here